### ISSUE TITLE
Allowing components to be SSR-friendly by changing pseudo class ":first-child" to ":first-of-type"

### ISSUE THREAD

**Issue by cheton** [trendmicro-frontend/tonic-ui#158]
A warning message reported by Next.js that the pseudo class `:first-child` is potentially unsafe when doing server-side rendering. Try changing it to `:first-of-type`.

![image](https://user-images.githubusercontent.com/447801/84344912-9474e680-abde-11ea-89af-1cc4a430e276.png)

Here are the component list that use the pseudo class ":first-child". It's suggested that we replace it with ":first-of-type" to make components to be SSR-friendly.

```sh
src/Checkbox/styles.js:      '& > :first-child': {
src/Input/styles.js:    '&:not(:first-child)': {
src/InputGroupAppend/styles.js:    '& > *:first-child': notFirstChildStyle,
src/InputGroupAppend/styles.js:    '&:not(:last-child) > *:first-child': notLastChildStyle,
src/InputGroupPrepend/styles.js:    '& > *:first-child': notLastChildStyle,
src/InputGroupPrepend/styles.js:    '&:not(:first-child) > *:first-child': notFirstChildStyle,
src/Modal/ModalFooter.js:        '&:first-child': {
src/Modal/ModalBody.js:        '&:first-child': {
```

**Comment by albertshala** [comment]
Hi @cheton, In Narnia, we have also seen this issue happen when using checkbox and modal components as well, for the modal component in particular, the above error happens only the first time the click event to open the modal occurs, all subsequent events to open or close the modal do not produce any error. 

Note: The error is hidden when build in production mode via webpack.

![Screen Shot 2020-10-29 at 5 48 44 PM](https://user-images.githubusercontent.com/888071/97636261-3a4eb380-1a0f-11eb-9544-d80f966fda94.png)

![issues-with-tonic-modal](https://user-images.githubusercontent.com/888071/97635782-46864100-1a0e-11eb-8dab-d8e18e6643a1.png)

**Comment by cheton** [comment]
It will be fixed in PR https://github.com/trendmicro-frontend/styled-ui/pull/253

**Comment by cheton** [comment]
Published [@trendmicro/react-styled-ui@0.9.3](https://github.com/trendmicro-frontend/styled-ui/releases/tag/%40trendmicro%2Freact-styled-ui%400.9.3) to NPM

https://www.npmjs.com/package/@trendmicro/react-styled-ui/v/0.9.3

**Comment by cheton** [comment]
See https://github.com/emotion-js/emotion/issues/1105#issuecomment-557726922 for details. 

Add `emotion-disable-server-rendering-unsafe-selector-warning-please-do-not-use-this-the-warning-exists-for-a-reason` to ignore warning messages ([packages/cache/src/stylis-plugins.js](https://github.com/emotion-js/emotion/blob/f3c51e8aca287170e43110bee2e4527eacfcada4/packages/cache/src/stylis-plugins.js#L125))

#### Option 1

```js
css`
  :first-child /* emotion-disable-server-rendering-unsafe-selector-warning-please-do-not-use-this-the-warning-exists-for-a-reason */ {
    color: hotpink;
  }
`
```

#### Option 2

```js
import { CacheProvider } from '@emotion/core'
import createCache from '@emotion/cache'

const myCache = createCache()
myCache.compat = true

<CacheProvider value={myCache}>
  <App/>
</CacheProvider>
```

#### Option 3

```js
import { cache } from 'emotion'

<CacheProvider value={cache}>
  <App/>
</CacheProvider>
```
